lib/pg/connection.rb

```diff
@@ -49,8 +49,7 @@ def self::parse_connect_args( *args )
 		if args.length == 1
 			case args.first
 			when URI, /\A#{URI::ABS_URI_REF}\z/
-				uri = URI(args.first)
-				options.merge!( Hash[URI.decode_www_form( uri.query )] ) if uri.query
+				uri = args.first
```

Comment:
I think it's better to have `uri` either always be `URI` or always be `String`. We don't use the structure of `URI` later on, so converting to `String` seems appropriate here.

```ruby
uri = String(args.first)
```